services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSpec.groovy

```diff
@@ -0,0 +1,463 @@
+        [srcSwitchValidateInfo, dstSwitchValidateInfo].each {
+            verifyRuleSectionsAreEmpty(it, ["missing", "excess"])
+        }
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        and: "Delete excess meters on the src and dst switches"
+        srcSwitchgetCreatedMeterIds.each { northbound.deleteMeter(srcSwitch.dpId, it) }
+        dstSwitchgetCreatedMeterIds.each { northbound.deleteMeter(dstSwitch.dpId, it) }
+
+        then: "Check that the switch validate request returns empty sections"
+        Wrappers.wait(WAIT_OFFSET) {
+            def srcSwitchValidateInfoAfterDelete = northbound.switchValidate(srcSwitch.dpId)
+            def dstSwitchValidateInfoAfterDelete = northbound.switchValidate(dstSwitch.dpId)
+            [srcSwitchValidateInfoAfterDelete, dstSwitchValidateInfoAfterDelete].each {
+                verifyRuleSectionsAreEmpty(it)
+                verifyMeterSectionsAreEmpty(it)
+            }
+        }
+    }
+
+    def "Switch validate is able to detect rule info into the 'missing' section"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.addFlow(flowHelper.randomFlow(srcSwitch, dstSwitch))
+
+        and: "Delete created rules on the srcSwitch"
+        northbound.deleteSwitchRules(srcSwitch.dpId, DeleteRulesAction.IGNORE_DEFAULTS)
+
+        then: "Rule info is moved into the 'missing' section on the srcSwitch"
+        def srcSwitchCreatedCookies = getCookiesWithMeter(srcSwitch.dpId)
+        def dstSwitchCreatedCookies = getCookiesWithMeter(dstSwitch.dpId)
+        def createdCookies = srcSwitchCreatedCookies + dstSwitchCreatedCookies
+
+        def srcSwitchValidateInfo = northbound.switchValidate(srcSwitch.dpId)
+        srcSwitchValidateInfo.rules.missing.containsAll(createdCookies)
+        verifyRuleSectionsAreEmpty(srcSwitchValidateInfo, ["proper", "excess"])
+
+        and: "Rule info is NOT moved into the 'missing' section on the dstSwitch"
+        def dstSwitchValidateInfo = northbound.switchValidate(dstSwitch.dpId)
+        dstSwitchValidateInfo.rules.proper.containsAll(createdCookies)
+        verifyRuleSectionsAreEmpty(dstSwitchValidateInfo, ["missing", "excess"])
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "Check that the switch validate request returns empty sections"
+        Wrappers.wait(WAIT_OFFSET) {
+            def srcSwitchValidateInfoAfterDelete = northbound.switchValidate(srcSwitch.dpId)
+            def dstSwitchValidateInfoAfterDelete = northbound.switchValidate(dstSwitch.dpId)
+            [srcSwitchValidateInfoAfterDelete, dstSwitchValidateInfoAfterDelete].each {
+                verifyRuleSectionsAreEmpty(it)
+                verifyMeterSectionsAreEmpty(it)
+            }
+        }
+    }
+
+    def "Able to get empty switch validate information from the intermediate switch(flow contain > 2 switches)"() {
```

Comment:
I think we should redesign the above tests (non-meter tests) to work with intermediate switches. This single test for transit switch is not enough. Should test for missing/excess rules too. Please keep in mind that these test will most likely be updated with `switch synchronize` in the future. And we should definitely check whether switch synchronize is able to deal with missing transit rules properly

Reply:
redesigned